extensions/spark/kyuubi-spark-lineage/src/test/scala/org/apache/spark/sql/SparkListenerExtenstionTest.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.nio.file.Files
+
+import org.apache.spark.SparkConf
+
+trait SparkListenerExtensionTest {
+  protected val catalogImpl: String
+  protected def format: String = if (catalogImpl == "hive") "hive" else "parquet"
+
+  protected lazy val spark: SparkSession = {
+    val metastore = {
+      val path = Files.createTempDirectory("hms")
```

Comment:
nit: create a temporary directory via `org.apache.spark.util.Utils.Utils.createTempDir()`, like we did in the `KyuubiSparkSQLExtensionTest` class, then the directory will be automatically deleted by the shutdown hook.

https://github.com/apache/incubator-kyuubi/blob/1642de3f1a2b303e6c74002a995912038fc1f860/extensions/spark/kyuubi-extension-spark-common/src/test/scala/org/apache/spark/sql/KyuubiSparkSQLExtensionTest.scala#L78-L93